File: 프로그래머스/lv2/131701. 연속 부분 수열 합의 개수/연속 부분 수열 합의 개수.py
def is_prime(n):
    
    for i in range(2, n // 2 + 1):
        if n % i == 0:
            return False
    
    return True

def solution(elements):
    
    numlist = set()
    
    for i in range(1, len(elements)+1):
        for j in range(len(elements)):
            num = elements[j:i+j]
            if len(num) < i:
                num += elements[:i - len(num)]
            numlist.add(sum(num))
    
    return len(numlist)
    
    
            

File: 프로그래머스/2/131701. 연속 부분 수열 합의 개수/연속 부분 수열 합의 개수.java
import java.util.*;
class Solution {
    
    static TreeSet<Integer> set;
    public int solution(int[] elements) {
        int answer = 0;
        
        // Arrays.sort(elements);
        set = new TreeSet<>();
        
        for(int i = 1; i <= elements.length; i++) {
            calc(i, elements);
        }
        
        // System.out.println(set);
        return set.size();
    }
    
    static void calc(int size, int[] elements) {

        for(int i = 0; i < elements.length; i++) {
            int num = elements[i];
            for(int j = i + 1; j < i + size; j++) {
                num += elements[j % elements.length];
            }
            set.add(num);
        }
    }
}